mitxgraders/formulagrader/intervalgrader.py

```diff
@@ -0,0 +1,231 @@
+"""
+intervalgrader.py
+
+Grader for intervals consisting of two formula entries, as well
+as enclosing brackets.
+"""
+from __future__ import print_function, division, absolute_import, unicode_literals
+
+from voluptuous import Required, Schema, Any, All, Length
+
+from mitxgraders.baseclasses import AbstractGrader
+from mitxgraders.stringgrader import StringGrader
+from mitxgraders.listgrader import SingleListGrader, demand_no_empty
+from mitxgraders.formulagrader.formulagrader import FormulaGrader, NumericalGrader
+from mitxgraders.exceptions import ConfigError, InvalidInput, MissingInput
+from mitxgraders.helpers.validatorfuncs import text_string
+
+class IntervalGrader(SingleListGrader):
+    """
+    IntervalGrader grades a pair of numbers describing an interval.
+    Both the interval bracketing and the numbers are graded.
+    The following are examples of possible entries:
+    * [1, 2)
+    * {0, 1+pi]
+
+    Answers should be provided as a list of four entries:
```

Comment:
I see that `IntervalGrader` subclasses `SingleListGrader`. Still, this seemed to me like an unexpected way to specify answers when grading an interval.

Thought: Is it worth having answers accept `[a, b)`, and just coercing to this form with a `schema_expect` if this form is needed internally?

(Would save from documenting two separate forms of the answer, too.)

Reply:
There's a reason to have all four entries separated in a list (or at least, the ability to do so) - so you can give partial credit/messages on incorrect brackets etc (the full ItemGrader format is available to each entry).

Regardless, now that the other PR is merged, I've added the other way to specify the answer: as a string that you would expect from students.